### PR TITLE
[MM-61477] Fixed exif rotates images width and height is not correctly calculated

### PR DESCRIPTION
**Pull request by fxnm**
#### Summary

Added additional rotation to the size based on the EXIF rotation of the image.
Because the in the current behavior the image size will wrong (height and width swapped), resulting in a cropped image on mobile.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-mobile/issues/8308
Jira: https://mattermost.atlassian.net/browse/MM-61477

#### Release Note

```release-note
Fixed EXIF rotated image previews not having the correct size.
```

### CODE REVIEW

**Review by fmartingr — CHANGES_REQUESTED** [review]
LGTM! This seems simple enough and do not require modifying the image. Thanks for your contribution!

Can we add one or some test cases to ensure this get's modified the way is intended?

File: server/channels/app/post_metadata_test.go [hunk 1]
```diff
@@ -2750,6 +2750,70 @@ func TestParseImages(t *testing.T) {
...
+			},
+		},
+		"jpg-8": {
+			FileName: "orientation_test_8.jpeg",
+			Expected: &model.PostImage{
+				Width:  2860,
+				Height: 1578,
+				Format: "jpeg",
+			},
+		},
```
Comment (fmartingr) [inline server/channels/app/post_metadata_test.go:2816]: Could we add some cases for PNG as well?

**Comment by fmartingr** [comment]
> Any updates?

It seems we have some flaky tests around. Let me check if those were fixed in master first.

**Review by hmhealey — APPROVED** [review]
Thanks for the PR, @fxnm! This looks great!

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**